### PR TITLE
Fixing AM RPC port binding issue

### PR DESCRIPTION
**Pull request by arde171**
Fixing AM RPC port allocation to avoid port binding issue.

### CODE REVIEW

**Comment by oliverhu** [comment]
how would this solve the problem ?

**Comment by goyalankit** [comment]
> how would this solve the problem ?

AM was using a random number to allocate port. We can use serversocket with port 0 to get OS to allocate a port.

**Comment by oliverhu** [comment]
@goyalankit  I see.. that was dumb, I thought we used port 0 to get OS

File: tony-core/src/main/java/com/linkedin/tony/ApplicationMaster.java [hunk 1]
```diff
@@ -385,7 +385,10 @@ private boolean prepare() throws IOException {
 
     // Setup application RPC server
     String amHostname = Utils.getCurrentHostName();
-    applicationRpcServer = setupRPCService(amHostname);
+    ServerSocket amRpcSocket = new ServerSocket(0);
+    int amRpcPort = amRpcSocket.getLocalPort();
+    amRpcSocket.close();
```
Comment (oliverhu) [inline tony-core/src/main/java/com/linkedin/tony/ApplicationMaster.java]: move this setupRPCService() ?

Comment (arde171) [reply tony-core/src/main/java/com/linkedin/tony/ApplicationMaster.java]: @oliverhu  ServerSocket may throw IOException https://docs.oracle.com/javase/8/docs/api/java/net/ServerSocket.html
So, we have to handle it cautiously. If we move this code to setupRPCService() then 
we need to handle throw exception there. Is it OK?

Comment (arde171) [reply tony-core/src/main/java/com/linkedin/tony/ApplicationMaster.java]: @oliverhu  As discussed, I moved the port allocation code inside the ApplicationRpcServer constructor.

**Pull request merged**